### ISSUE TITLE
*Critical OSS Vulnerability:* spring-web@5.3.20

### ISSUE THREAD

**Issue by tloubrieu-jpl** [NASA-PDS/registry-api#148]
*Critical OSS Vulnerability:*
### pkg:maven/org.springframework/spring-web@5.3.20
1 Critical, 0 Severe, 0 Moderate, 0 Unknown vulnerabilities have been found across 1 dependencies

<details>
  <summary><b>Components</b></summary><br/>
  <ul>
      <details>
        <summary><b>pkg:maven/org.springframework/spring-web@5.3.20</b></summary>
        <ul>
  <details>
    <summary><b>CRITICAL Vulnerabilities (1)</b></summary><br/>
<ul>

> #### [CVE-2016-1000027] CWE-502: Deserialization of Untrusted Data
> Pivotal Spring Framework through 5.3.16 suffers from a potential remote code execution (RCE) issue if used for Java deserialization of untrusted data. Depending on how the library is implemented within a product, this issue may or not occur, and authentication may be required. NOTE: the vendor&#39;s position is that untrusted data is not an intended use case. The product&#39;s behavior will not be changed because some users rely on deserialization of trusted data.
>
> **CVSS Score:** 9.8
>
> **CVSS Vector:** CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H
>
> **CWE:** CWE-502

</ul>
    </details>
        </ul>
      </details>
  </ul>
</details>

Reply with *"**@sonatype-lift help**"* for more info.
Reply with *"**@sonatype-lift ignore**"* to tell Liftbot to leave out the above finding from this PR.
Reply with *"**@sonatype-lift ignoreall**"* to tell Liftbot to leave out all the findings from this PR and from the status bar in Github.

When talking to Liftbot, you need to **refresh** the page to see its response. [Click here](https://help.sonatype.com/lift/talking-to-lift) to get to know more about Liftbot commands.

---

Was this a good recommendation?
[ [🙁 Not relevant](https://www.sonatype.com/lift-comment-rating?comment=277153478&lift_comment_rating=1) ] - [ [😕 Won't fix](https://www.sonatype.com/lift-comment-rating?comment=277153478&lift_comment_rating=2) ] - [ [😑 Not critical, will fix](https://www.sonatype.com/lift-comment-rating?comment=277153478&lift_comment_rating=3) ] - [ [🙂 Critical, will fix](https://www.sonatype.com/lift-comment-rating?comment=277153478&lift_comment_rating=4) ] - [ [😊 Critical, fixing now](https://www.sonatype.com/lift-comment-rating?comment=277153478&lift_comment_rating=5) ]

_Originally posted by @sonatype-lift[bot] in https://github.com/NASA-PDS/registry-api/pull/147#discussion_r895072364_

**Comment by tloubrieu-jpl** [comment]
No upgrade is available for this package at the time of creation of this ticket. 

Need to confirm with @al-niessner if this vulnerability applies to the registry-api (launched jar vs launched as war). See https://help.sonatype.com/docs/important-announcements/find-and-fix-springshell

If not applicable we'll close this ticket as 'wontfix'. If applicable but not workaround is available yet we'll have the ticket in `icebox`.

**Comment by al-niessner** [comment]
@tloubrieu-jpl 

It has the problem that it will/can execute any URL (no scrubbed) but same is true of all previous versions -- the API docs require it be able to do it. If Apache is put on the front end as a proxy and used to scrub the URL, then this problem is moot.

**Comment by tloubrieu-jpl** [comment]
Apache does not clean anything in front of the API. Either we need to sanitize all of our API parameters (which would make sense, ticket https://github.com/NASA-PDS/registry-api/issues/219) and/or we upgrade to the new version 6.0.X available, which requires to upgrade to jdk 17 (https://github.com/NASA-PDS/registry-api/issues/218).

**Comment by tloubrieu-jpl** [comment]
This ticket is blocked by one of these tickets, listed above

**Comment by jordanpadams** [comment]
Hopefully fixed by this: https://github.com/NASA-PDS/registry-api/pull/201

**Comment by tloubrieu-jpl** [comment]
This ticket requires upgrade to jdk17

**Comment by jordanpadams** [comment]
blocked by #218

**Comment by jordanpadams** [comment]
Status: All the endpoints work but swagger UI broke. That has now been fixed, and something else has broken. API docs endpoint is broken.
